core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -56,7 +56,7 @@ object Partitioner {
    */
   def defaultPartitioner(rdd: RDD[_], others: RDD[_]*): Partitioner = {
     val bySize = (Seq(rdd) ++ others).sortBy(_.partitions.size).reverse
-    for (r <- bySize if r.partitioner.isDefined) {
+    for (r <- bySize if r.partitioner.isDefined && r.partitions.nonEmpty) {
       return r.partitioner.get
     }
```

Comment:
We are dealing with two different members.. We are checking whether r.partitoner.isDefined and returning only when r.partitions.nonEmpty.  There can be cases when r.partitioner is not defined but r.partitions can be non empty.